compiler/dyno/lib/parsing/parsing-queries.cpp

```diff
@@ -598,6 +598,27 @@ bool idIsParenlessFunction(Context* context, ID id) {
   return idIsParenlessFunctionQuery(context, id);
 }
 
+static const bool& idIsFieldQuery(Context* context, ID id) {
+  QUERY_BEGIN(idIsFieldQuery, context, id);
+
+  bool result = false;
+  AstTag tag = idToTag(context, id);
```

Comment:
Just checked and agree it can probably be rewritten, I'll give it a try~

```cpp
if (auto ast = astForIDQuery(context, id))
  if (auto var = ast->toVariable())
    result = var->isField();
```

Reply:
Right, but if we are doing an incremental recompile, it is more likely that idToTag is there and going to be reused, where astForId is comparatively unlikely to be reused. Anyway it is probably moot here & it's something we can have more insight to when we do performance studies of incremental recompiles.